packages/components/col/src/col.ts

```diff
@@ -24,6 +32,17 @@ export const colProps = buildProps({
   span: {
     type: Number,
     default: 24,
+    validator: (value) => {
+      const isValid = Number.isInteger(value) && value >= 0 && value <= 24
+      if (!isValid) {
+        isNumber(value) &&
+          debugWarn(
+            COL_COMPONENT_NAME,
+            `The "span" prop expects an integer between 0 and 24, but received: "${value}".`
+          )
+      }
+      return isValid
+    },
```

Comment:
Maybe the validation should not be placed here, otherwise two warning messages will appear, which seems a bit redundant.